client/client.go

```diff
@@ -267,6 +269,80 @@ func vsockDial(host, port string) (net.Conn, string, error) {
 
 }
 
+func contains(elems []string, v string) bool {
+	for _, s := range elems {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+// check that mdns response has all required attributes
+func mdns_required(src []string, req map[string][]string) bool {
+	txt := make(map[string]string)
+	for element := range src {
+		s := strings.SplitN(src[element], "=", 2)
+		txt[s[0]] = s[1]
+	}
+	for k, _ := range req {
+		if !contains(req[k], txt[k]) {
+			return false
+		}
+	}
+	return true
+}
+
+// lookup based on hostname, return resolved host, port, network, and error
+// uri currently supported dnssd://domain/_service._network/instance?reqkey=reqvalue
+// default for domain is local, first path element is _ncpu._tcp, and instance is wildcard
+// can omit to underspecify, e.g. dnssd:?arch=arm64 to pick any arm64 cpu server
+func mdns_lookup(uri string) (string, string, string, error) {
+	var (
+		hostname string
+		port     string
+		network  string
+		err      error
+	)
+
+	// determine if we want to do mdns
+	u, err := url.Parse(uri)
+	if (err == nil) && (u.Scheme == "dnssd") {
```

Comment:
you can save an entire layer of indent for the following ~30 lines of code if you structure the code as:
if err != nil {
  // return code
}
if u.Scheme != "dnssd") {
  return "", "", "", nil // Nothing to do here...
}
// rest of code.

Reply:
yeah, I waffled between the two, happy to side with not having the indent.